src/Microsoft.Health.Fhir.Core/Features/Conformance/SystemConformanceProvider.cs

```diff
@@ -14,36 +14,43 @@
 using Microsoft.Health.Fhir.Core.Extensions;
 using Microsoft.Health.Fhir.Core.Features.Conformance.Models;
 using Microsoft.Health.Fhir.Core.Features.Definition;
-using Microsoft.Health.Fhir.Core.Messages.Search;
+using Microsoft.Health.Fhir.Core.Features.Validation;
+using Microsoft.Health.Fhir.Core.Messages.CapabilityStatement;
 using Microsoft.Health.Fhir.Core.Models;
 
 namespace Microsoft.Health.Fhir.Core.Features.Conformance
 {
     public sealed class SystemConformanceProvider
-        : ConformanceProviderBase, IConfiguredConformanceProvider, INotificationHandler<SearchParametersUpdated>, IDisposable
+        : ConformanceProviderBase, IConfiguredConformanceProvider, INotificationHandler<RebuildCapabilityStatement>, IDisposable
```

Comment:
I don't really agree that a Notification should be implied in this direction, for these points:

- If `SystemConformanceProvider` is listening/responding to notifications, then I think it should contain the responsibility of knowing when it should regenerate, based on the type of events it is interested in. i.e. `SearchParametersUpdated`
- If the notification is so specific (i.e. `RebuildCapabilityStatement`), and the generating class knew it needed to do this, it would make more sense for it to explicitly call  something like IConfiguredConformanceProvider.Regerenate()

Reply:
I agree and I prefer first approach. I was trying to get it pass tests first, and then refactor into normal state.
Didn't anticipate change of few line in json to be 37 files -_-'